src/server/search/search_family.cc

```diff
@@ -46,28 +46,38 @@ bool IsValidJsonPath(string_view path) {
   return !ec;
 }
 
-pair<size_t, search::VectorSimilarity> ParseVectorFieldInfo(CmdArgParser* parser,
-                                                            ConnectionContext* cntx) {
+search::SchemaField::VectorParams ParseVectorParams(CmdArgParser* parser) {
   size_t dim = 0;
-  search::VectorSimilarity sim = search::VectorSimilarity::L2;
+  auto sim = search::VectorSimilarity::L2;
+  size_t capacity = 1000;
 
+  bool use_hnsw = parser->ToUpper().Next().Case("HNSW", true).Case("FLAT", false);
```

Comment:
:exploding_head:

Reply:
Better than 5+ lines `if {} else if {} else reply` 😼

Reply:
Don't get me wrong, I'm not saying it's bad. It's great, readable, no complaints.
I was just blown away that this is C++ and not, idk, Python :rofl:

Reply:
> Don't get me wrong, I'm not saying it's bad. It's great, readable, no complaints.
I was just blown away that this is C++ and not, idk, Python

How do you feel about `std::ranges` then ? :stuck_out_tongue: